etcdserver/server.go

```diff
@@ -612,8 +615,18 @@ func (s *EtcdServer) run() {
 			sched.Schedule(f)
 		case leases := <-expiredLeaseC:
 			go func() {
-				for _, l := range leases {
-					s.LeaseRevoke(context.TODO(), &pb.LeaseRevokeRequest{ID: int64(l.ID)})
+				// Increases throughput of exipired leases deletion process through parallization
+				c := make(chan struct{}, goroutinesCountForProcessingExpiredLeases)
+				for _, lease := range leases {
+					select {
+					case c <- struct{}{}:
+					case <-s.done:
+						return
+					}
+					go func() {
+						s.LeaseRevoke(context.TODO(), &pb.LeaseRevokeRequest{ID: int64(lease.ID)})
+						<-c
```

Comment:
@fanminshi the issue isn't garbage collection, it's the possibility of a goroutine leak from waiting on `c` forever. Looking at it again, I see the `close` isn't needed after all because there'll be a token on `c` for every `Revoke`, so there's no need to `close`...